File: Python/0388-m-longest-absolute-file-path.py
class Solution:
    def lengthLongestPath(self, input):
        """
        :type input: str
        :rtype: int
        """
        stack = [(-1, 0)]
        max_len = 0
        for p in input.split('\n'):
            depth = p.count('\t')
            p = p.replace('\t','')
            while stack and depth <= stack[-1][0]:
                stack.pop()
            if '.' not in p:
                stack.append((depth, len(p) + stack[-1][1] + 1))
            else:
                max_len = max(max_len, len(p) + stack[-1][1])
        return max_len

File: leetcode/C++/0388-m-longest-absolute-file-path.cpp
class Solution {
public:
    int lengthLongestPath(string input) {
        istringstream ss(input);
        string cur;
        int res = 0;
        unordered_map<int, int> pathLen;
        while (getline(ss, cur, '\n')) {
            auto depth = cur.find_last_of("\t");
            string name = (depth == string::npos) ? cur : cur.substr(depth + 1);
            if (cur.find(".") != string::npos) {
                res = max(res, pathLen[depth - 1] + (int)name.size());
            } else {
                pathLen[depth] = pathLen[depth - 1] + name.size() + 1;
            }
        }
        return res;
    }
};